### PR TITLE
Support editing config file from the command line

### LINKED ISSUE #164: CLI command to set config options
As discussed in https://github.com/djmitche/taskchampion/discussions/152

It would help with writing good user-guides to have a CLI which can set config options, maybe something like

    task config set sync-server-origin http://example.com

or maybe more locally specific

    task sync set-url http://example.com

or something along those lines.

Justification being that "run this command to set the server" is a lot friendly than "edit the config at [platform dependent location], then in this section, add the line [...etc]"

Git is a good reference for this - e.g the commands for setting name/email are presented in error messages from git itself, remote URL's are presented by hosting services like Github.

### PR DESCRIPTION
**Pull request by djmitche**
Fixes #164.

### CODE REVIEW

**Comment by dbr** [comment]
This looks like a great change :partying_face: The docs for configuring the sync server seem a lot easier to follow along given this change

File: cli/src/invocation/cmd/config.rs [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    match config_operation {
+        ConfigOperation::Set(key, value) => {
+            let filename = settings.set(&key, &value)?;
+            write!(w, "Set configuration value ")?;
+            w.set_color(ColorSpec::new().set_bold(true))?;
+            write!(w, "{}", &key)?;
+            w.set_color(ColorSpec::new().set_bold(false))?;
+            write!(w, " in ")?;
+            w.set_color(ColorSpec::new().set_bold(true))?;
+            write!(w, "{:?}.\n", filename)?;
```
Comment (dbr) [inline cli/src/invocation/cmd/config.rs:19]: Hmm, not sure I agree with the clippy lint here (maybe a separate `write!(w, "\n")` would be clearer and appease the lint?)

However I wonder if it's worth investigating a neater way to do the coloured text formatting as `write!(w, "Set configuration value {} in {:?}", ...)` is drastically easier to read than the version with the coloured output

Comment (dbr) [reply cli/src/invocation/cmd/config.rs:19]: Split this into #229 as it's unrelated to this change really

Comment (djmitche) [reply cli/src/invocation/cmd/config.rs:19]: I'm OK with appeasing clippy, though :)

**Review by dbr — APPROVED** [review]
LGTM!

**Pull request merged**